src/sentry/options/manager.py

```
@@ -80,16 +81,10 @@ def set(self, key, value, coerce=True):
         # Enforce immutability if value is already set on disk
         assert not (opt.flags & FLAG_PRIORITIZE_DISK and settings.SENTRY_OPTIONS.get(key)), '%r cannot be changed at runtime because it is configured on disk' % key
 
-        if not self._value_is_of_type(opt.type, value):
-            if not coerce:
-                raise TypeError('got %r, expected %r' % (_type(value), opt.type))
-            else:
-                # TODO(dcramer): implement more explicit coercion error
-                # with custom types
-                try:
-                    value = opt.type(value)
-                except (ValueError, TypeError):
-                    raise TypeError('Unable to coerce %r to %r' % (_type(value), opt.type))
+        if coerce:
+            value = opt.type(value)
+        elif not opt.type.test(value):
+            raise TypeError('got %r, expected %r' % (_type(value), opt.type))
```

should this raise InvalidTypeError